src/components/signup/SignupState.js

```diff
@@ -462,6 +465,25 @@ const Signup = ({ navigation }: { navigation: any, screenProps: any }) => {
     return prevRoute
   }
 
+  //check if email/mobile was used to register before and offer user to login instead
+  const checkExisting = async searchBy => {
+    const { exists, fullName, provider: foundOtherProvider } = await userExists2(searchBy).catch(e => {})
+    log.debug('checking userAlreadyExist', { exists, fullName, foundOtherProvider })
+    let selection = 'signup'
+    if (exists || foundOtherProvider) {
+      selection = await showAlreadySignedUp(
+        torusProvider,
+        exists,
+        foundOtherProvider,
+        searchBy.email ? 'email' : 'mobile',
+      )
+      if (selection === 'signin') {
+        return navigation.navigate('Auth', { screen: 'signin' })
+      }
+    }
+    return selection
+  }
+
   const done = async (data: { [string]: string }) => {
```

Comment:
1. wrap to useCallback
2. reformat a bit code (at least empty lines between code blocks)